patches/server/0807-Multiple-Entries-with-Scoreboards.patch

```diff
@@ -0,0 +1,125 @@
++                anyAdded = true;
++            }
++        }
++
++        if (anyAdded) {
++            this.sendAll(ClientboundSetPlayerTeamPacket.createMultiplePlayerPacket(team, players, ClientboundSetPlayerTeamPacket.Action.ADD));
++            this.setDirty();
++            return true;
++        } else {
++            return false;
++        }
++    }
++    // Paper end
++
+     @Override
+     public void removePlayerFromTeam(String playerName, PlayerTeam team) {
+         super.removePlayerFromTeam(playerName, team);
+@@ -99,6 +124,17 @@ public class ServerScoreboard extends Scoreboard {
+         this.setDirty();
+     }
+ 
++    // Paper start
++    public void removePlayersFromTeam(java.util.Collection<String> players, PlayerTeam team) {
++        for (String playerName : players) {
++            super.removePlayerFromTeam(playerName, team);
++        }
++
++        this.sendAll(ClientboundSetPlayerTeamPacket.createMultiplePlayerPacket(team, players, ClientboundSetPlayerTeamPacket.Action.REMOVE));
++        this.setDirty();
++    }
++    // Paper end
++
+     @Override
+     public void onObjectiveAdded(Objective objective) {
+         super.onObjectiveAdded(objective);
+diff --git a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
+index 2b87a652798cb632fe76bf20e9e7f8cb8bfb3b7b..666edc7f0e0cd814cbc7525eaf7ec2ed486e3dac 100644
+--- a/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
++++ b/src/main/java/org/bukkit/craftbukkit/scoreboard/CraftTeam.java
+@@ -226,6 +226,16 @@ final class CraftTeam extends CraftScoreboardComponent implements Team {
+         scoreboard.board.addPlayerToTeam(entry, team);
+     }
+ 
++    // Paper start
++    @Override
++    public void addEntries(java.util.Collection<String> entries) throws IllegalStateException, IllegalArgumentException {
++        Validate.notNull(entries, "Entries cannot be null");
++        CraftScoreboard scoreboard = checkState();
++
++        ((net.minecraft.server.ServerScoreboard) scoreboard.board).addPlayersToTeam(entries, team);
++    }
++    // Paper end
++
+     @Override
+     public boolean removePlayer(OfflinePlayer player) throws IllegalStateException, IllegalArgumentException {
+         Validate.notNull(player, "OfflinePlayer cannot be null");
+@@ -245,6 +255,26 @@ final class CraftTeam extends CraftScoreboardComponent implements Team {
+         return true;
+     }
+ 
++    // Paper start
++    @Override
++    public boolean removeEntries(java.util.Collection<String> entries) throws IllegalStateException, IllegalArgumentException {
++        Validate.notNull(entries, "Entry cannot be null");
++        CraftScoreboard scoreboard = this.checkState();
++
++        boolean anyRemoved = false;
++        for (String entry : entries) {
++            if (!anyRemoved && this.team.getPlayers().contains(entry)) {
++                anyRemoved = true;
```

Comment:
remove the anyRemoved boolean and just put the removal here and return afterwards